### ISSUE TITLE
[question] Data-grid expanded row

### ISSUE THREAD

**Issue by chitsutote** [mui/mui-x#9384]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Hi, I am using the free version of x-data-grid. I am trying to have a expanded row when I click on the cell.
I used `getDetailPanelContent` prop to render my expanded row, but nothing happened.

I also attempted to use the `useApiRef` hook to programmatically toggle the expanded row by calling `apiRef.current.toggleDetailPanel`, but unfortunately, it didn't work. Upon further investigation, I realized that the `toggleDetailPanel` function is not accessible in the free version, as mentioned in the documentation.

I'm wondering if the DetailPanel feature is not available in the free version. If that's the case, it's confusing to have the `getDetailPanelContent` prop available in the free version. Is there any workaround or hack that I can use to achieve the expanded row functionality with the free data grid? Or do I need to build my own grid using a table component?

Any guidance or suggestions would be greatly appreciated. Thank you!

### Expected behavior 🤔

Should show expanded row when I click on the cell

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by romgrk** [comment]
From what I can see in [our docs](https://mui.com/x/react-data-grid/master-detail/), the detail panels are only available for premium and pro users.

The blue square indicates pro features, the yellow one premium features:
![image](https://github.com/mui/mui-x/assets/1423607/38ee2291-e5ba-49aa-aa3b-9f300f4c55b0)


> it's confusing to have the getDetailPanelContent prop available in the free version

Do you mean the typings include that prop?

> Is there any workaround or hack that I can use to achieve the expanded row functionality with the free data grid?

I don't think so, but you could use an alternative such as using a modal.

**Comment by chitsutote** [comment]
> From what I can see in [our docs](https://mui.com/x/react-data-grid/master-detail/), the detail panels are only available for premium and pro users.
> 
> The blue square indicates pro features, the yellow one premium features: ![image](https://user-images.githubusercontent.com/1423607/246858499-38ee2291-e5ba-49aa-aa3b-9f300f4c55b0.png)
> 
> > it's confusing to have the getDetailPanelContent prop available in the free version
> 
> Do you mean the typings include that prop?
No. I mean I can use this prop to render custom detail panel content with free data grid component.
I think that is confusing because it dont have expanded row.

> 
> > Is there any workaround or hack that I can use to achieve the expanded row functionality with the free data grid?
> 
> I don't think so, but you could use an alternative such as using a modal.
